src/lang/modifyAst/addFillet.ts

```diff
@@ -231,6 +322,89 @@ export function addFillet(
   return new Error('Unsupported extrude type.')
 }
 
+function getPathToNodeOfFilletLiteral(
+  pathToExtrudeNode: PathToNode,
+  extrudeDeclarator: VariableDeclarator,
+  tag: string
+): PathToNode {
+  let pathToFilletObj: any
+  let inFillet = false
+
+  traverse(extrudeDeclarator.init, {
+    enter(node, path) {
+      if (node.type === 'CallExpression' && node.callee.name === 'fillet') {
+        inFillet = true
+      }
+      if (inFillet && node.type === 'ObjectExpression') {
+        if (!hasTag(node, tag)) return false
+        pathToFilletObj = getPathToRadiusLiteral(node, path)
+      }
+    },
+    leave(node) {
+      if (node.type === 'CallExpression' && node.callee.name === 'fillet') {
+        inFillet = false
+      }
+    },
+  })
+  let indexOfPipeExpression = pathToExtrudeNode.findIndex(
+    (path) => path[1] === 'PipeExpression'
+  )
+
+  indexOfPipeExpression =
+    indexOfPipeExpression === -1
+      ? pathToExtrudeNode.length
+      : indexOfPipeExpression
+
+  return [
+    ...pathToExtrudeNode.slice(0, indexOfPipeExpression),
+    ...pathToFilletObj,
+  ]
+}
+
+function hasTag(node: ObjectExpression, tag: string): boolean {
+  return node.properties.some((prop) => {
+    if (prop.key.name === 'tags' && prop.value.type === 'ArrayExpression') {
+      return prop.value.elements.some(
+        (element) => element.type === 'Identifier' && element.name === tag
+      )
+    }
+    return false
+  })
+}
+
+function getPathToRadiusLiteral(node: ObjectExpression, path: any): any {
+  let pathToFilletObj = path
+  node.properties.forEach((prop, index) => {
+    if (prop.key.name === 'radius') {
+      pathToFilletObj.push(
+        ['properties', 'ObjectExpression'],
+        [index, 'index'],
+        ['value', 'Property']
+      )
+    }
+  })
+  return pathToFilletObj
+}
+
+function getFilletTag(existingFilletCall: CallExpression): string | null {
```

Comment:
`export function getTagFromCallExpression(` in `src/lang/std/sketch.ts` has a very similar function signature, just wanted to double check if these are doing different things?

Reply:
it has different purpose and structure (for segments) and has to be heavily modified in order to work with fillets. 

```
export function getTagFromCallExpression(
  callExp: CallExpression
): string | Error {
  if (callExp.callee.name === 'close') return getTag(1)(callExp)
  if (callExp.callee.name in sketchLineHelperMap) {
    const { getTag } = sketchLineHelperMap[callExp.callee.name]
    return getTag(callExp)
  }
  return new Error(`"${callExp.callee.name}" is not a sketch line helper`)
}
```

so we would have add another if statement like: `if (callExp.callee.name === 'fillet')` and then copy paste the logic of current `getFilletTag` in it.

Or you mean using the same logic as over there and do something similar?
would be something like this:
```
function getTagFromCallExpression2(callExp: CallExpression): string | Error {
  return getTag()(callExp)
}

function getTag(index = 0) {
  return (callExp: CallExpression) => {
    if (callExp.type !== 'CallExpression') {
      return new Error('Not a CallExpression')
    }

    // Access the correct argument based on the provided index
    const arg = callExp.arguments?.[index]
    if (!arg || arg.type !== 'ObjectExpression') {
      return new Error('Expected an ObjectExpression as the argument')
    }

    const properties = (arg as ObjectExpression).properties
    const tagsProperty = properties.find((prop) => prop.key.name === 'tags')
    
    if (!tagsProperty || tagsProperty.value.type !== 'ArrayExpression') {
      return new Error('No valid tags property found')
    }

    const elements = (tagsProperty.value as ArrayExpression).elements
    if (elements.length === 0 || elements[0].type !== 'Identifier') {
      return new Error('No valid tag identifier found')
    }

    return elements[0].name // Return the name of the first tag (identifier)
  }
}
```